### PR TITLE
Use Windows mode if build # is <= 19041

### LINKED ISSUE #91898: Adopt text wrapping on conpty
When https://github.com/microsoft/terminal/issues/405 hits Windows we will be able to disable the special "windows mode" the terminal enables which guesses wrapped lines.

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #91898

per https://github.com/microsoft/terminal/issues/405#issuecomment-592096770

### CODE REVIEW

**Comment by meganrogge** [comment]
Correctly detects that my system needs windows mode. Wish I could verify on a higher windows build that it works correctly

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -901,7 +908,8 @@ export class TerminalService implements ITerminalService {
 			this._terminalShellTypeContextKey,
 			this._terminalAltBufferActiveContextKey,
 			this._configHelper,
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:910]: considered adding this to shellLaunchConfig, but figured that wouldn't add much

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 2]
```diff
@@ -204,14 +206,18 @@ export class TerminalService implements ITerminalService {
...
+    	// Wait up to 5 seconds for profiles to be ready so it's assured that we know the actual
 		// default terminal before launching the first terminal. This isn't expected to ever take
 		// this long.
 		this._profilesReadyBarrier = new AutoOpenBarrier(5000);
 		this._refreshAvailableProfiles();
 	}
 
+	private async _initRequiresWindowsMode(): Promise<void> {
+		this._requiresWindowsMode = await this._offProcessTerminalService?.requiresWindowsMode() || false;
+	}
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:219]: I think we'd be better off evaluating this every time when the process gets created. Right now this wouldn't work if you toggled conpty/winpty and would also cause issues when creating local terminals in remote workspaces.

I mentioned the traits idea yesterday but wasn't too keen on requiring a round trip to access it, here's a better idea:

```ts
IPtyService.createProcess(...): Promise<{
  persistentProcessId: number,
  requiresWindowsMode?: boolean
}>
```

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:895]: shell launch config generally shouldn't be modified once the process gets created as it's purpose is to create/recreate the process, not for checking its result

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts [hunk 3]
```diff
@@ -299,10 +300,10 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 			dispose(this._processListeners);
 		}
 		this._processListeners = [
-			newProcess.onProcessReady((e: { pid: number, cwd: string }) => {
+			newProcess.onProcessReady(async (e: IProcessTraits) => {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts:303]: async not needed?

File: src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts [hunk 4]
```diff
@@ -79,14 +79,15 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 	private _shellLaunchConfig?: IShellLaunchConfig;
 	private _dimensions: ITerminalDimensions = { cols: 0, rows: 0 };
 	private _isScreenReaderModeEnabled: boolean = false;
+	private _hasRemoteAuthority: boolean = false;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts:82]: Don't need a prop?

File: src/vs/workbench/test/browser/workbenchTestServices.ts [hunk 5]
```diff
@@ -1607,6 +1607,7 @@ export class TestLocalTerminalService implements ILocalTerminalService {
 	processBinary(id: number, data: string): Promise<void> { throw new Error('Method not implemented.'); }
 	updateTitle(id: number, title: string): Promise<void> { throw new Error('Method not implemented.'); }
 	updateIcon(id: number, icon: string, color?: string): Promise<void> { throw new Error('Method not implemented.'); }
+	requiresWindowsMode(): Promise<boolean> { throw new Error('Method not implemented.'); }
```
Comment (Tyriar) [inline src/vs/workbench/test/browser/workbenchTestServices.ts:1610]: No longer needed?

File: src/vs/platform/terminal/common/terminal.ts [hunk 6]
```diff
@@ -418,6 +418,8 @@ export interface ITerminalLaunchError {
 	code?: number;
 }
 
+export interface IProcessTraits { pid: number, cwd: string, requiresWindowsMode?: boolean }
```
Comment (Tyriar) [inline src/vs/platform/terminal/common/terminal.ts:421]: Let's call this IProcessReadyEvent for consistency with the other events, also properties on new line and `;` over `,`

**Review by Tyriar — APPROVED** [review]

**Pull request merged**